tests/test_schedule.py

```diff
@@ -165,7 +165,7 @@ def test_get_common_stations(self):
         assert '2' in (common_stations["3"])
 
     def test_get_negative_rotations(self):
-        """Check if the single rotation '1' with a negative soc is found """
+        """Check if rotations '1' and '11' with negative SOCs are found """
```

Comment:
The test tries making clear that rotation "1" is obviously negative. Its is not clear why 11 should be negative too. 
This happens because of 11 and 1 share the same vehicle and the soc of -120 can not be recharged until the next day. My suggestion is calculating a slightly negative soc by using the mileage and distance or moving the great distance to the last rotation taking place. This way it will not change other rotations
the "bad" assigning of vehicles should not be caught here.